llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp

```diff
@@ -1750,6 +1752,88 @@ void lowerGlobalsToVector(Module &M) {
 
 } // namespace
 
+static void checkSLMInit(Module &M) {
+  SmallPtrSet<const Function *, 8u> Callers;
+  bool Kernel_Has_slm_init = false;
```

Comment:
nit: I'm not used to seeing this code style for var names, maybe KernelHasSLMInit?